### PR TITLE
Use valid git credentials when multiple are provided

### PR DESCRIPTION
**Pull request by arthurchaloin**
See #1668 for details about the issue. Here I will discuss about the fix implementation.

If the given URL works out of the box or if a single token is available in the environment, the behaviour remains unchanged.
However if there is multiple tokens available, it will perform a `git.verifyAuth` call for each of them, and return the first of them that is working.
If none of them are working, it will return the URL as is.

This behaviour is tested in `integration.test.js`. It isn't tested in `get-git-auth-url.test.js` because I need gitbox functionalities to test it.

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -656,3 +657,43 @@ test('Hide sensitive environment variable values from the logs', async (t) => {
...
+        GIT_TERMINAL_PROMPT: 0,
+      },
+      branch: {name: 'master'},
+      options: {repositoryUrl: 'http://toto@localhost:2080/git/test-auth.git'},
+    }),
+    authUrl
+  );
+});
+
+test('Use the reposirotry URL as is if none of the given git credentials are valid', async (t) => {
```
Comment (gr2m) [inline test/integration.test.js:681]:
```suggestion
test('Use the repository URL as is if none of the given git credentials are valid', async (t) => {
```

**Review by gr2m — COMMENTED** [review]
Great work! Just a small typo. I would have fixed it myself but don't have write access to your fork

**Comment by arthurchaloin** [comment]
Fixed the typo and the conflicts, ready to merge!

**Review by gr2m — APPROVED** [review]
Thanks 👍🏼

**Pull request merged**